x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/counter_rate.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { FormattedIndexPatternColumn, ReferenceBasedIndexPatternColumn } from '../column_types';
+import { IndexPatternLayer } from '../../../types';
+import { checkForDateHistogram, dateBasedOperationToExpression, hasDateField } from './utils';
+import { OperationDefinition } from '..';
+
+const ofName = (name: string) => {
+  return i18n.translate('xpack.lens.indexPattern.CounterRateOf', {
+    defaultMessage: 'Counter rate of {name}',
+    values: { name },
+  });
+};
+
+export type CounterRateIndexPatternColumn = FormattedIndexPatternColumn &
+  ReferenceBasedIndexPatternColumn & {
+    operationType: 'counter_rate';
+  };
+
+export const counterRateOperation: OperationDefinition<
+  CounterRateIndexPatternColumn,
+  'fullReference'
+> = {
+  type: 'counter_rate',
+  priority: 2,
```

Comment:
This priority value won't make a big difference because we won't directly suggest these operations, but I still think we should reduce the priority of reference-based operations to 1. This will make sure that they are sorted to the end of the suggestion list, after buckets and metrics.